frontend/pages/hosts/details/HostQueryReport/HostQueryReport.tsx

```diff
@@ -0,0 +1,416 @@
+    //     report_clipped: false,
+    //     last_fetched: null,
+    //     results: [],
+    //   },
+    //   {
+    //     name: 'test query',
+    //     description: "a great query",
+    //     query: "SELECT * FROM users",
+    //     discard_data: false,
+    //     inverval: 20,
+    //   },
+    // ],
+
+    // // nothing to report (B)
+    // [
+    //   {
+    //     host_name: "Haley's Macbook Air",
+    //     report_clipped: false,
+    //     last_fetched: "2021-01-01T00:00:00.000Z",
+    //     results: [],
+    //   },
+    //   {
+    //     name: 'test query',
+    //     description: "a great query",
+    //     query: "SELECT * FROM users",
+    //     discard_data: false,
+    //     inverval: 20,
+    //   },
+    // ],
+
+    // // report clipped (C)
+    // [
+    //   {
+    //     host_name: "Haley's Macbook Air",
+    //     report_clipped: true,
+    //     last_fetched: "2021-01-01T00:00:00.000Z",
+    //     results: [],
+    //   },
+    //   {
+    //     name: "test query",
+    //     description: "a great query",
+    //     query: "SELECT * FROM users",
+    //     interval: 20,
+    //     discard_data: false,
+    //   },
+    // ],
+
+    // // reroute (local setting)
+    // [
+    //   {
+    //     host_name: "Haley's Macbook Air",
+    //     report_clipped: false,
+    //     last_fetched: "2021-01-01T00:00:00.000Z",
+    //     results: [
+    //       {
+    //         columns: {
+    //           username: "user1",
+    //           email: "e@mail",
+    //         },
+    //       },
+    //     ],
+    //   },
+    //   {
+    //     name: 'test query',
+    //     description: "a great query",
+    //     query: "SELECT * FROM users",
+    //     discard_data: true,
+    //     inverval: 20,
+    //   },
+    // ],
+
+    false,
+    null,
+  ];
+
+  // const {
+  //   data: hqrResponse,
+  //   isLoading: hqrLoading,
+  //   error: hqrError,
+  // } = useQuery<IGetHQRResponse, Error>(
+  //   [hostId, queryId],
+  //   () => hqrAPI.load(hostId, queryId),
+  //   {
+  //     refetchOnMount: false,
+  //     refetchOnReconnect: false,
+  //     refetchOnWindowFocus: false,
+  //   }
+  // );
+
+  // const {
+  //   isLoading: queryLoading,
+  //   data: queryResponse,
+  //   error: queryError,
+  // } = useQuery<IGetQueryResponse, Error, ISchedulableQuery>(
+  //   ["query", queryId],
+  //   () => queryAPI.load(queryId),
+  //   {
+  //     enabled: !!queryId,
+  //     refetchOnMount: false,
+  //     refetchOnReconnect: false,
+  //     refetchOnWindowFocus: false,
+  //   }
+  // );
+
+  // TODO - remove mock loading state
+  const queryLoading = false;
+
+  const isLoading = queryLoading || hqrLoading;
+
+  const {
+    host_name: hostName,
+    report_clipped: reportClipped,
+    last_fetched: lastFetched,
+    results,
+    // TODO - remove below casting, just for testing
+  } = (hqrResponse || {}) as Partial<IGetHQRResponse>;
+
+  const rows = results?.map((row) => row.columns) ?? [];
```

Comment:
I think a code comment would be helpful here to explain why this is necessary (i.e. what is the shape of the incoming results, why to they need to be processed in this way).

Reply:
I guess the shape of `IHQRResult` just isn't very intuitive to me. Why are columns getting mapped to rows?